packages/app/lib/ui/widgets/image_provider.dart

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:image_picker/image_picker.dart';
+
+class MeliCameraProvider extends StatefulWidget {
+  const MeliCameraProvider(this.child, {Key? key}) : super(key: key);
+  final Widget child;
+
+  @override
+  MeliCameraProviderState createState() => MeliCameraProviderState();
+}
+
+class MeliCameraProviderState extends State<MeliCameraProvider> {
+  final _imagePicker = ImagePicker();
+  String? _retrieveDataError;
+  List<File> images = [];
+
+  List<File> getAll() {
+    return images;
+  }
+
+  bool isEmpty() {
+    return (images.length == 0);
+  }
+
+  void addAll(List<File> newImages) {
+    setState(() {
+      newImages.addAll(images);
+      images = newImages;
+    });
+  }
+
+  void removeAll() {
+    setState(() {
+      images = [];
+    });
+  }
+
+  void removeAt(int index) {
+    setState(() {
+      List<File> newImages = [];
+      newImages.addAll(images);
+      newImages.removeAt(index);
+      images = newImages;
+    });
+  }
+
+  Future<void> retrieveLostData() async {
+    final LostDataResponse response = await _imagePicker.retrieveLostData();
+    if (response.isEmpty) {
+      return;
+    }
+    if (response.file != null) {
+      addAll([File(response.file!.path)]);
+    } else {
+      _retrieveDataError = response.exception!.code;
+      print('CameraProvider error: ${_retrieveDataError}');
+    }
+  }
+
+  // Pick photos from the gallery.
+  Future<void> pickFromGallery() async {
+    List<XFile> imageFiles =
+        await _imagePicker.pickMultiImage(imageQuality: 50);
+
+    addAll(imageFiles.map((file) {
+      return File(file.path);
+    }).toList());
+  }
+
+  // Capture a photo from the camera.
+  Future<void> capturePhoto() async {
+    ImageSource source = ImageSource.camera;
+    if (!_imagePicker.supportsImageSource(source)) {
+      print('Camera source not supported');
+      source = ImageSource.gallery;
+    }
+
+    XFile? imageFile = await _imagePicker.pickImage(
+        source: ImageSource.camera,
+        imageQuality: 50,
```

Comment:
Do we want to increase the image quality? If it is used for scientific purposes? How about 80?

Reply:
Yeh, will bump to 80!